app/scripts/controllers/permissions/specifications.js

```diff
@@ -176,6 +188,45 @@ export const getPermissionSpecifications = ({
         }
       },
     },
+
+    [PermissionNames.permittedChains]: {
+      permissionType: PermissionType.Endowment,
+      targetName: PermissionNames.permittedChains,
+      allowedCaveats: [CaveatTypes.restrictNetworkSwitching],
+      subjectTypes: [SubjectType.Website],
+
+      factory: (permissionOptions, requestData) => {
+        if (!requestData.approvedChainIds) {
+          throw new Error(
+            `${PermissionNames.permittedChains}: No approved networks specified.`,
+          );
+        }
+
+        const caveats = permissionOptions.caveats ?? [
+          CaveatFactories[CaveatTypes.restrictNetworkSwitching](
+            requestData.approvedChainIds,
+          ),
+        ];
```

Comment:
I'm not sure if this is right. For `eth_accounts`, for example, we recently started permitting caller-specified caveats, which would appear here as `permissionOptions.caveats`. However, the accounts we should actually set are determined by the UI, and appear as `requestData.approvedAccounts` (see above).

Similarly, I believe we should permit but ignore caller-specified caveats in here, and just use `requestData.approvedChainIds`. Caller-specified caveats can still inform the confirmation in the UI, as was the intention with the recent change to `eth_accounts`.

Reply:
It appears that the UI is already prepared to handle this [ref](https://github.com/MetaMask/metamask-extension/blob/2619ccb6f02998350651a273e4abb7869599e58e/ui/components/app/permission-page-container/permission-page-container.component.js#L156-L160), so it should just be a matter of this:
```suggestion
        const caveats = [
          CaveatFactories[CaveatTypes.restrictNetworkSwitching](
            requestData.approvedChainIds,
          ),
        ];
```

Reply:
done here: https://github.com/MetaMask/metamask-extension/pull/24415/commits/80fbb98731c7206a3a11fd57bd62c6c8c3989a3d